vmm/src/api/http_endpoint.rs

```diff
@@ -646,3 +647,42 @@ impl EndpointHandler for VmAddVsock {
         }
     }
 }
+
+// /api/v1/vm.balloon-resize handler
+pub struct VmBalloonResize {}
+
+impl EndpointHandler for VmBalloonResize {
+    fn handle_request(
+        &self,
+        req: &Request,
+        api_notifier: EventFd,
+        api_sender: Sender<ApiRequest>,
+    ) -> Response {
+        match req.method() {
+            Method::Put => {
+                match &req.body {
+                    Some(body) => {
+                        // Deserialize into a VmConfig
+                        let resize_data: u64 = match serde_json::from_slice(body.raw())
```

Comment:
Why are you using JSON encoding APIs (here and in the remote) if you're only sending a single u64. Please define a struct with a single member to handle this.